### PR TITLE
(1) search users by name or email (2) OAuth re log in after token expires (3) get domain for current customer

### PR DESCRIPTION
**Pull request by wenyi-guo**

### CODE REVIEW

File: src/main/webapp/scripts/user-script.js [hunk 1]
```diff
@@ -54,19 +61,77 @@ function addOrgUnits2Data(ous){
         }
         break;
     }
-    
+}
+
+function addOrgUnits2DataSearch(ous){
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js]: maybe rename to addOrgUnitsToDataSearch?

File: src/main/webapp/scripts/user-script.js [hunk 2]
```diff
@@ -1,7 +1,14 @@
 var params = JSON.parse(localStorage.getItem('oauth2-test-params'));
 var token = params['access_token'];
+var domain = localStorage.getItem('domain');
 var flatdata = []
 var data = {}
+var input;
+
+function userOnload(){
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js:8]: is this function needed if its just calling two other functions and only called once, can those other two functions just be called where userOnload() is called?

File: src/main/webapp/scripts/user-script.js [hunk 3]
```diff
@@ -101,9 +166,10 @@ function addUsertoOUByPath(node, path, user){
     }
 }
 
-// change value of each OU to the number of users in the OU
+// change value of each OU to the number of users in the OU, 
+// value is 1 more than the number of users to prevent empty orgUnits display issue 
 function incrementUserCount (node){
-    var value = node.data['users'].length;
+    var value = node.data['users'].length + 1;
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js]: what is value in this context? can the variable be renammed to something more descriptive

File: src/main/webapp/scripts/user-script.js [hunk 4]
```diff
@@ -154,7 +220,7 @@ function visualize() {
 
         node.append("rect")
             .attr("id", function(d) { return d.data.id; })
-            .attr("fill", d => d === root ? "#fff" : d.children ? "#ccc" : "#ddd")
+            .attr("fill", d => d === root ? "#fff" : d.children ? "#99bbff" : "#ccddff")
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js]: maybe comment saying what these colors are ?

File: src/main/webapp/scripts/user-script.js [hunk 5]
```diff
@@ -474,4 +538,57 @@ function visualizeUser(userData, htmlid){
...
\ No newline at end of file
+}
+
+// sidebar functionality, including getting domain name, searchbar
+// filter (todo)
+function sidebar(){
+    domainElement = document.getElementById("domain-name");
+    domainElement.innerText = domain;
+    searchField = document.getElementById("user-search-input");
+    searchBtn = document.getElementById("user-search-btn");
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js]: probably best to rename this variable to searchButton

File: src/main/webapp/scripts/user-script.js [hunk 6]
```diff
@@ -433,8 +497,8 @@ function visualizeUser(userData, htmlid){
     // append the svg object to the body of the page
     // appends a 'group' element to 'svg'
     // moves the 'group' element to the top left margin
-    var str = "#" + htmlid;
-    var svg = d3.select(str).append("svg")
+    var strr = "#" + htmlid;
```
Comment (ajs11893) [inline src/main/webapp/scripts/user-script.js]: if this is not used anywhere else except directly below the line below can be rewritten as 
`d3.select( "#" + htmlid).append("svg")`
or rename the variable to something more descriptive

**Review by ajs11893 — APPROVED** [review]
Still not sure why OUs with no users are not consistently sized. Also some variable names should be changed to be a little easier to understand. Overall looks good

File: src/main/webapp/scripts/user-script.js (hunk 1 above)
Comment (wenyi-guo) [reply src/main/webapp/scripts/user-script.js]: changes in new commit

**Pull request merged**